### PR TITLE
Fix activity names not translated in interruption alerts

### PR DESCRIPTION
**Pull request by Qrox**
#### Summary
SUMMARY: Bugfixes "Fix activity names not translated in interruption alerts"

#### Purpose of change
Activity names were not translated in interruption alerts. This PR fixes it.

#### Describe the solution
Use `translation` to store and translate activity verbs. The previous comparison operators of `translation` are changed to ordinary functions, and new comparison operators that compares the raw data are created. This is because during json data loading `assign` uses the `==` operator to check if the loaded variable has changed, so `operator ==` should compare the precise underlying data instead of the translated string. The `std::string current_activity` member of `npc` is changed to `activity_id current_activity_id`, since it is written to the saves, so the ID needs to be saved instead of the translation. By using `translation`, activity verbs in json can be specified with contexts now, so the translation documentation is updated to reflect that.

### CODE REVIEW

File: src/optional.h [hunk 1]
```diff
@@ -234,6 +234,30 @@ class optional
         }
 };
 
+template<class T, class U>
+constexpr bool operator==( const optional<T> &lhs, const optional<U> &rhs )
+{
+    if( bool( lhs ) != bool( rhs ) ) {
```
Comment (AMurkin) [inline src/optional.h]: It probably should be
```cpp
    if( lhs.has_value() != rhs.has_value() ) {
```
Same for the `operator!=`.

Comment (Qrox) [reply src/optional.h]: `operator bool()` works the same as `has_value()`.

File: src/player_activity.h [hunk 2]
```diff
@@ -12,8 +12,9 @@
 
 #include "enums.h"
 #include "item_location.h"
-#include "string_id.h"
 #include "point.h"
+#include "string_id.h"
+#include "translations.h"
```
Comment (BevapDin) [inline src/player_activity.h]: You only need a forward declaration here, not the whole header.

File: src/optional.h (hunk 1 above)
Comment (BevapDin) [reply src/optional.h]: We generally don't use c-style or function-style casts, instead we use `static_cast`. But in this case `has_value` works the same.

File: src/optional.h [hunk 3]
```diff
@@ -234,6 +234,30 @@ class optional
...
+        return false;
+    } else if( !lhs ) {
+        return true;
+    } else {
+        return *lhs == *rhs;
+    }
+}
+
+template< class T, class U >
+constexpr bool operator!=( const optional<T> &lhs, const optional<U> &rhs )
```
Comment (BevapDin) [inline src/optional.h]: `operator!=` is typically implement with a single statement: ` return !operator==( lhs, rhs );` - this ensures the relation between `==` and `!=` is as one would expect them.

**Pull request merged**